src/tsickle_compiler_host.ts

```diff
@@ -105,7 +105,37 @@ export class TsickleCompilerHost implements ts.CompilerHost {
 
   constructor(
       private delegate: ts.CompilerHost, private tscOptions: ts.CompilerOptions,
-      private options: Options, private environment: TsickleHost) {}
+      private options: Options, private environment: TsickleHost) {
+    // ts.CompilerHost includes a bunch of optional methods.  If they're
+    // present on the delegate host, we want to delegate them.
+    if (this.delegate.getCancellationToken) {
+      this.getCancellationToken = () => this.delegate.getCancellationToken!();
+    }
+    if (this.delegate.getDefaultLibLocation) {
+      this.getDefaultLibLocation = () => this.delegate.getDefaultLibLocation!();
+    }
+    if (this.delegate.resolveModuleNames) {
+      this.resolveModuleNames = (moduleNames, containingFile) =>
+          this.delegate.resolveModuleNames!(moduleNames, containingFile);
```

Comment:
I think in each of these you can write it like

this.resolveModuleNames = this.delegate.resolveModuleNames!.bind(this.delegate);

that will let you omit all the parameter passing around.